File: src/ELibrary.Backend/LibraryShopEntities/Domain/Dtos/Shop/OrderResponse.cs
using LibraryShopEntities.Domain.Entities.Shop;

namespace LibraryShopEntities.Domain.Dtos.Shop
{
    public class OrderResponse
    {
        public int Id { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public int OrderAmount { get; set; }
        public decimal TotalPrice { get; set; }
        public string ContactClientName { get; set; } = default!;
        public string ContactPhone { get; set; } = default!;
        public string DeliveryAddress { get; set; }
        public DateTime DeliveryTime { get; set; }
        public OrderStatus OrderStatus { get; set; }
        public PaymentMethod PaymentMethod { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
        public ClientResponse Client { get; set; }
        public List<OrderBookResponse> OrderBooks { get; set; }
    }
}

File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/shop/order/orderResponse.ts
import { ClientResponse, DeliveryMethod, mapClientResponseToClient, mapOrderBookResponseToOrderBook, Order, OrderBookResponse, OrderStatus, PaymentMethod } from "../../../..";

export interface OrderResponse {
    id: number,
    createdAt: Date,
    updatedAt: Date,
    orderAmount: number,
    totalPrice: number,
    contactClientName: string,
    contactPhone: string,
    deliveryAddress: string,
    deliveryTime: Date,
    orderStatus: OrderStatus,
    paymentMethod: PaymentMethod,
    deliveryMethod: DeliveryMethod,
    client: ClientResponse,
    orderBooks: OrderBookResponse[]
}

export function mapOrderResponseToOrder(response: OrderResponse): Order {
    return {
        id: response?.id,
        createdAt: new Date(response?.createdAt),
        updatedAt: new Date(response?.updatedAt),
        orderAmount: response?.orderAmount,
        totalPrice: response?.totalPrice,
        contactClientName: response?.contactClientName,
        contactPhone: response?.contactPhone,
        deliveryAddress: response?.deliveryAddress,
        deliveryTime: new Date(response?.deliveryTime),
        orderStatus: response?.orderStatus,
        paymentMethod: response?.paymentMethod,
        deliveryMethod: response?.deliveryMethod,
        client: mapClientResponseToClient(response?.client),
        orderBooks: response?.orderBooks.map(x => mapOrderBookResponseToOrderBook(x))
    }
}